csvcubed/csvcubed/models/codelistconfig/code_list_config.py

```diff
@@ -0,0 +1,86 @@
+from typing import Optional, List, Tuple, Dict
+from dataclasses import dataclass, field
+from pathlib import Path
+
+from csvcubedmodels.dataclassbase import DataClassBase
+
+from csvcubed.models.cube.qb.catalog import CatalogMetadata
+from csvcubed.utils.json import load_json_document
+from csvcubed.models.cube.qb.components import NewQbConcept
+from csvcubed.readers.catalogmetadata.v1.catalog_metadata_reader import (
+    metadata_from_dict,
+)
+
+CODE_LIST_CONFIG_DEFAULT_URL = "https://purl.org/csv-cubed/code-list-config/v1.0"
+
+
+@dataclass
+class CodeListConfigSort(DataClassBase):
+    """Model for representing the sort object in code list config."""
+
+    by: str
+    method: str
+
+
+@dataclass
+class CodeListConfigConcept(DataClassBase):
+    """Model for representing a code list concept in code list config."""
+
+    label: str
+    notation: str
+    parent_notation: Optional[str] = field(default=None)
```

Comment:
@robons This value is necessary for defining the `parent_code` for the `NewQbConcepts` in the `new_qb_concepts` property in this dataclass.

Reply:
It isn't, take a look at the comment I made in that section. We definitely don't want `parent_notation` to be provided by external users given they've already defined the `children` - it's two ways to specify the same information that can only cause bugs.

Reply:
Ok makes sense - I will update accordingly.

Reply:
@robons Instead of defining the `tuple(parent_notation, concept)`, I've created a new method called `_get_parent_of_concept()`. This method returns the parent of the given concept if exists.

The reason for creating this method is that using `tuple(parent_notation, concept)` implies that we are still storing a `parent_notation` explicitly rather than finding the `parent_notation` from the `children`. What are your thoughts on this?

Reply:
I don't think it implies we're expecting the user to provide the information since the interface won't have parent notation on it. 

I suspect your approach will be less efficient given that you'll have to iterate over all items (in the worst case) in the code list every time you want to find out what its parent is.